File: examples/c-help/src/show-help.c
#include <stdlib.h>
#include <stdio.h>
#include <unistd.h>

#include "help.h"
#include "man.h"

void print(const char* name) 
{
   const struct man_content* c = man_get_content (name);
   if (c == NULL) 
     {
       c = man_get_help_content (name);
       if (c == NULL)
         {
           printf ("FAIL: No help for '%s'\n", name);
           exit (1);
         }
     }

   if (write (STDOUT_FILENO, c->content, c->size) != c->size)
     {
       exit (1);
     }
}


int main(int argc, char** argv)
{
   if (argc <= 1) 
     {
       printf("Help names:\n");
       for (int i = 0; i < help_names_length; i++)
         {
           printf("  %s\n", help_names[i]);
         }
       printf("Man pages:\n");
       for (int i = 0; i < man_names_length; i++)
         {
           printf("  %s\n", man_names[i]);
         }
        return 0;
     }
   for (int i = 1; i < argc; i++)
     {
        print (argv[i]);
     }
   return 0;
}

File: examples/go-help/show-help.go
package main

import (
   "os"
   "fmt"
   "go-help/help"
   "go-help/man"
)

func print(name string) {
   c1 := man.Get_content(name)
   if (c1 != nil) {
      fmt.Println(string(c1.Content))
      return;
   }

   c2 := help.Get_content(name);
   if (c2 == nil) {
       fmt.Println("FAIL: No configuration file ", name)
       return
   }
   fmt.Println(string(c2.Content))
}

func main() {
   if (len(os.Args) <= 1) {
      fmt.Println("Help names:")
      for _, name := range help.Names {
          fmt.Println("  ", name);
      }

      fmt.Println("Man pages:")
      for _, name := range man.Names {
          fmt.Println("  ", name);
      }
      return
   }

   for _, arg := range os.Args[1:] {
       print(arg)
   }
}
